File: Day2/uniqueopfromip.py
# output = input().split()
output = input()
output = [int(num) for num in output]
print(output)
def uniqueopfromip(numbers):
    uniquenumber=set()
    for num in numbers:
        if numbers.count(num)==1:
            uniquenumber.add(num)
    return list(uniquenumber)

print(uniqueopfromip(output))

# input_numbers = list(map(int, input("Enter numbers separated by spaces: ").split()))

# num= [3,5,2,3,5]
# for nums in num:
#     print(num.count(nums))
# print(num.count(5))

File: Day2/uniqueopfromip.c
#include <stdio.h>
#include <stdbool.h>

// Function to check if a character is a digit
bool isDigit(char c) {
    return (c >= '0' && c <= '9');
}

// Function to find unique digits in a string
void findUniqueDigits(char* str) {
    int count[10] = {0}; // Array to store the count of each digit
    int len = strlen(str);

    // Count the occurrences of each digit
    for (int i = 0; i < len; i++) {
        if (isDigit(str[i])) {
            int digit = str[i] - '0';
            count[digit]++;
        }
    }

    // Print the unique digits
    printf("Unique digits: ");
    for (int i = 0; i < 10; i++) {
        if (count[i] == 1) {
            printf("%d ", i);
        }
    }
    printf("\n");
}

int main() {
    char input[100]; // Assuming a maximum input length of 100 characters

    printf("Enter a string of digits (without spaces): ");
    scanf("%s", input);

    findUniqueDigits(input);

    return 0;
}
